src/plugins/intel_gpu/src/graph/impls/cpu/shape_of.cpp

```diff
@@ -60,9 +58,13 @@ struct shape_of_impl : public typed_primitive_impl<shape_of> {
             OPENVINO_THROW("[GPU] Couldn't execute shape_of operation: unsupported output data type (", output_dt , ")");
         }
 
-        ev->set();
-
-        return ev;
+        if (events.size() > 1) {
+            return stream.group_events(events);
```

Comment:
Could you please clarify, why do we need to propagate input events to the next operations in case of shape_of? Wouldn't it cause negative performance impact?

Reply:
Previously, `shape_of` did not wait for dependent events, causing accuracy issues.  GPU plugin's synchronization logic assumes that when an event completed, all previous events have already finished. Prior to this PR, the `shape_of` made a custom event and set it as finished without checking the status of its dependent events. As a result, the users of `shape_of` were assuming that all previous events had completed and proceeding, when in fact there were still events that hadn't completed. So, I updated it to return the dependent events to ensure that `shape_of`'s users are executed after `shape_of`'s dependent events are finished